bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/CaConfigReferenceProvider.java

```diff
@@ -0,0 +1,60 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2020 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.services;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.caconfig.resource.ConfigurationResourceResolver;
+import org.osgi.service.component.annotations.Component;
+
+import com.adobe.cq.wcm.core.components.config.HtmlPageItemsConfig;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.reference.Reference;
+import com.day.cq.wcm.api.reference.ReferenceProvider;
+
+/**
+ * Provides Context Aware Configuration references for a given resource.
+ */
+@Component(
+        service = ReferenceProvider.class
+)
+public class CaConfigReferenceProvider implements ReferenceProvider {
+
+    private static final String NN_SLING_CONFIGS = "sling:configs";
+    private static final String CA_CONFIG_REFERENCE_TYPE = "caconfig";
+
+    /**
+     * The @{@link ConfigurationResourceResolver} service.
+     */
+    @org.osgi.service.component.annotations.Reference
+    private ConfigurationResourceResolver configurationResourceResolver;
+
+    @Override
+    public List<Reference> findReferences(Resource resource) {
+        List<Reference> references = new ArrayList<>();
+        Resource configResource = configurationResourceResolver.getResource(resource, NN_SLING_CONFIGS, HtmlPageItemsConfig.class.getName());
```

Comment:
We should expand this to cover all context-aware configurations shipped with Core Components: datalayer and PDF Viewer.

Reply:
addressed in commit https://github.com/adobe/aem-core-wcm-components/commit/2adf80cbc76e5d6213db21cbff1c3f45a9dddcaf

Reply:
It would be nice to have a generic approach for this so this reference provider doesn't need an update each time we add a new CaConfig interface.

Reply:
good idea. I don't have time to work on it now and it's a change that can be implemented later. I'll merge as it is now and we'll tackle it later.